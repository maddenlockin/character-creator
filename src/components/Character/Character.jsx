import React from 'react'

export default function Character({top, middle, bottom}) {
    return (
        <section style={{ display: "flex", flexDirection: "column" }}>
            <article
                className="top"
                style={{
                    backgroundImage: `url(./${top}-chick-head.png)`,
                    height: 200,
                    width: 200,
                    backgroundSize: "cover",
                    backgroundRepeat: "no-repeat",
                }}
            ></article>
            <article
                className="middle"
                style={{
                    backgroundImage: `url(./${middle}-belly.png)`,
                    height: 150,
                    width: 150,
                    backgroundSize: "cover",
                    backgroundRepeat: "no-repeat",
                }}
            ></article>
            <article
                className="bottom"
                style={{
                    backgroundImage: `url(./${bottom}-feet.png)`,
                    height: 150,
                    width: 150,
                    backgroundSize: "cover",
                    backgroundRepeat: "no-repeat",
                }}
            ></article>
        </section>
    );
}
