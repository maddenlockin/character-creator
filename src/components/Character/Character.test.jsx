import { screen, render } from "@testing-library/react";
import Character from "./Character";

it("should render character", async () => {
    const { container } = render(
        <Character top="church" middle="wrinkles" bottom="cat" />
    );
    const element = await screen.findAllByRole("article");

    expect(element[0]).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});
