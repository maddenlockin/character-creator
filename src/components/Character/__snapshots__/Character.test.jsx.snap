// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render character 1`] = `
<div>
  <section
    style="display: flex; flex-direction: column;"
  >
    <article
      class="top"
      style="background-image: url(./church-chick-head.png); height: 200px; width: 200px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      class="middle"
      style="background-image: url(./wrinkles-belly.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    />
    <article
      class="bottom"
      style="background-image: url(./cat-feet.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    />
  </section>
</div>
`;
